### DISCUSSION TITLE
Welcome to sdp-seismic-unix Discussions!

### DISCUSSION THREAD

**Discussion by hadi-tim** [hadi-tim/sdp-seismic-unix#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

I created this modest page on Seismic Data Processing using Seismic Unix to share it with students and seismic enthusiast people. I hope you will find it useful.
The page is opened to all your comments and suggestions.

**Comment by wannabesuccessguy** [comment]
hi, im a freshgraduate geophysics currently trying to enter seismic processing role, do you have an entry level project reccomendation for seismic data processing using seismic unix?

**Comment by hadi-tim** [comment]
Hello,

My sincere apologies for my late reply, unfortunately no, I don't have any
specific project regarding seismic data processing, but you can follow my
tutorial on basics of SDP and try to do the same.

Kind regards,
Hadi.

**Comment by bvermeulen** [comment]
Hi Hadi, I have start to go through the tutorial, but got stuck at: 

```After appying the geometry we can notice that headers are correctly updated including the offset and X 1 Y coordinates.```

I cannot find the su command to actually apply the geometry to the data, please can you clarify.

Thanks, Bruno Vermeulen.

**Reply by bvermeulen** [reply]
I think the answer is:
```sushw < line_001.su infile=myheaders.bin key=sx,sy,selev,sstat,gx,gy,gelev,gstat,offset > line_001e.su```

**Comment by bvermeulen** [comment]
To have the correct offset values annotated above the shot gather with geometry applied, you need to set a `d2` value, like in this shell script named `shot_gather.sh`:
```
suwind key=ep min=$1 max=$1 < data/line_001e.su | suximage key=offset d2=25 cmap=hsv4 perc=90 \
   title="shot $1 after geometry" label1="Time (s)" label2="Offset (m)"
```
here `$1` is the ep shot number, for instance:
```
>./shot_gather.sh 200
```

The reason is that the `key` value is only read from the first trace and default interval may not be appropriate. See [su help](https://sepwww.stanford.edu/oldsep/cliner/files/suhelp/suximage.txt):
```
SUXIMAGE - X-windows IMAGE plot of a segy data set			

 suximage <stdin [optional parameters] | ...				

 Optional parameters:						 	

 n2=tr.ntr or number of traces in the data set (ntr is an alias for n2)

 d1=tr.d1 or tr.dt/10^6	sampling interval in the fast dimension	
   =.004 for seismic 		(if not set)				
   =1.0 for nonseismic		(if not set)				

 d2=tr.d2		sampling interval in the slow dimension		
   =1.0 		(if not set)					

 key=			key for annotating d2 (slow dimension)		
 			If annotation is not at proper increment, try	
 			setting d2; only first trace's key value is read

 f1=tr.f1 or tr.delrt/10^3 or 0.0  first sample in the fast dimension	

 f2=tr.f2 or tr.tracr or tr.tracl  first sample in the slow dimension	
   =1.0 for seismic		    (if not set)			
   =d2 for nonseismic		    (if not set)			

 verbose=0              =1 to print some useful information		

 tmpdir=	 	if non-empty, use the value as a directory path	
		 	prefix for storing temporary files; else if the	
	         	the CWP_TMPDIR environment variable is set use	
	         	its value for the path; else use tmpfile()	

 Note that for seismic time domain data, the "fast dimension" is	
 time and the "slow dimension" is usually trace number or range.	
 Also note that "foreign" data tapes may have something unexpected	
 in the d2,f2 fields, use segyclean to clear these if you can afford	
 the processing time or use d2= f2= to override the header values if	
 not.									

 See the ximage selfdoc for the remaining parameters.			
```

**Comment by hadi-tim** [comment]
I believe this is the case when you want to apply a nominal geometry, using
regular spasing.
In my case I applied the real geometry to data using the corresponding SPS
files (s, r and x).

Kind regards,
Hadi.

Le dim. 29 sept. 2024 à 05:03, Bruno Vermeulen ***@***.***> a
écrit :

**Reply by bvermeulen** [reply]
I believe this is not the case. I also applied the geometry from the SPS and got the same result as that you had shown where the offset annotation does not make sense, as they range from -3520 to -3240 over the full shot gather, at least I would expect positive offset values on the right side.

From the documentation:
```
 key=			key for annotating d2 (slow dimension)		
 			If annotation is not at proper increment, try	
 			setting d2; only first trace's key value is read
```

**Comment by hadi-tim** [comment]
Oooops, I didn't pay attention to my script, it was a typing error. I
eventually forgot to insert the distance element d2 to display correct
offsets. Thanks for spotting that.
Display has wrong offset values, but originally they are correct, please
see my screenshot below:
[image: image.png]


Le dim. 29 sept. 2024 à 05:03, Bruno Vermeulen ***@***.***> a
écrit :
